cli/switch_test.go

```diff
@@ -0,0 +1,144 @@
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	testCase struct {
+		name          string
+		args          []string
+		configExists  bool
+		configContent string
+	}
+
+	expectedOutput struct {
+		stdout string
+		stderr string
+	}
+)
+
+func TestSwitchCommand(t *testing.T) {
+	tcs := []struct {
+		in  testCase
+		out expectedOutput
+	}{
+		{
+			in: testCase{
+				name:         "Config file doesn't exist",
+				args:         []string{"org1"},
+				configExists: false,
+			},
+			out: expectedOutput{
+				stdout: "Please run `dispatch login` to login to Dispatch.\n",
+			},
+		},
+		{
+			in: testCase{
+				name:         "No arguments provided",
+				args:         []string{},
+				configExists: true,
+				configContent: `
+	# Warning = 'THIS FILE IS GENERATED. DO NOT EDIT!'
+	active = 'x-s-org'
+	
+	[Organizations]
+	[Organizations.x-s-org]
+	api_key = 'x'
+	`,
+			},
+			out: expectedOutput{
+				stdout: "Available organizations:\n- x-s-org\n",
+			},
+		},
+		{
+			in: testCase{
+				name:         "Switch to non-existing organization",
+				args:         []string{"random"},
+				configExists: true,
+				configContent: `
+	# Warning = 'THIS FILE IS GENERATED. DO NOT EDIT!'
+	active = 'x-s-org'
+	
+	[Organizations]
+	[Organizations.x-s-org]
+	api_key = 'x'
+	`,
+			},
+			out: expectedOutput{
+				stdout: "Organization 'random' not found\n\nAvailable organizations:\n- x-s-org\n",
+			},
+		},
+		{
+			in: testCase{
+				name:         "Switch to existing organization",
+				args:         []string{"x-s-org"},
+				configExists: true,
+				configContent: `
+	# Warning = 'THIS FILE IS GENERATED. DO NOT EDIT!'
+	active = 'x-s-org'
+	
+	[Organizations]
+	[Organizations.x-s-org]
+	api_key = 'x'
+	`,
+			},
+			out: expectedOutput{
+				stdout: "Switched to organization: x-s-org\n",
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.in.name, func(t *testing.T) {
+			// Set up the config file
+			configPath := setupConfig(t, tc.in)
+
+			// Set the DispatchConfigPath (global variable)
+			DispatchConfigPath = configPath
+
+			// Create buffer writer
+			stdout := &bytes.Buffer{}
+			stderr := &bytes.Buffer{}
+
+			// Create command in test
+			cmd := switchCommand()
+
+			// Set streams for command
+			cmd.SetOut(stdout)
+			cmd.SetErr(stderr)
+
+			// Set args for command
+			cmd.SetArgs(tc.in.args)
+
+			// Execute command
+			if err := cmd.Execute(); err != nil {
+				t.Fatalf("Received unexpected error: %v", err)
+			}
+
+			assert.Equal(t, tc.out.stdout, stdout.String())
+			assert.Equal(t, tc.out.stderr, stderr.String())
+		})
+	}
+}
+
+func setupConfig(t *testing.T, tc testCase) string {
+	tempDir, err := os.MkdirTemp("", "dispatch-test")
+	assert.NoError(t, err)
+	t.Cleanup(func() { os.RemoveAll(tempDir) })
+
+	configPath := filepath.Join(tempDir, "config.yaml")
+	if tc.configExists {
+		err = os.WriteFile(configPath, []byte(tc.configContent), 0600)
+		assert.NoError(t, err)
+	} else {
+		os.Remove(DispatchConfigPath)
```

Comment:
This feels like the kind of tests that will be brittle; e.g., if it gets run concurrently, access to the global value will cause unexpected behaviors.

I'd suggest we leave a comment at least warning about this, unless we have a better avenue for making it more robust.

Reply:
I copy and pasted the comment I left above in the "Potential Next Steps" section. I definitely agree that this is brittle. 

What do you think about the suggestion of a sort of IO interface the commands would take in? This would primarily allow mocking. 

A good example of this dependency injection is Kubectl:  https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/create/create_clusterrole.go#L73

Any other suggestions?

Reply:
Something like passing the globals to `cli.Main` maybe?

I don't think it's worth adding tons of infrastructure right now to solve it, but worth documenting our thought process in a comment, and maybe filing another issue to address later.

Reply:
Sgtm. Filed an issue to do something similar with all the other command's global dependencies: https://github.com/dispatchrun/dispatch/issues/61